components/bt/bluedroid/btc/profile/std/avrc/btc_avrc.c

```diff
@@ -250,6 +258,75 @@ static void handle_rc_disconnect (tBTA_AV_RC_CLOSE *p_rc_close)
 #endif
 }
 
+static void handle_rc_attributes_rsp ( tAVRC_MSG_VENDOR* vendor_msg){
+    uint8_t attr_count = vendor_msg->p_vendor_data[4];
+    int attr_index = 5;
+    int attr_length = 0;
+
+    //Check if there are any attributes
+    if(attr_count < 1) return;
+
+    esp_avrc_ct_cb_param_t param[attr_count];
+    memset(&param[0], 0, sizeof(esp_avrc_ct_cb_param_t) * attr_count);
+
+  for(int i = 0; i < attr_count; i++){
+    attr_length = (int) vendor_msg->p_vendor_data[7 + attr_index] | vendor_msg->p_vendor_data[6 + attr_index] << 8;
+
+    //Copy text to buffer, null terminate it
+    memcpy(meta_text_buffer[i], &vendor_msg->p_vendor_data[8 + attr_index], 0x7F & attr_length);
+    meta_text_buffer[i][0x7F & attr_length] = 0;
+
+    param[i].meta_rsp.attr_text = meta_text_buffer[i];
+
+    param[i].meta_rsp.attr_id = vendor_msg->p_vendor_data[3 + attr_index] |
+      vendor_msg->p_vendor_data[2 + attr_index] << 8 | vendor_msg->p_vendor_data[1 + attr_index] << 16 |
+      vendor_msg->p_vendor_data[attr_index] << 24;
+
+    btc_avrc_ct_cb_to_app(ESP_AVRC_CT_METADATA_RSP_EVT, &param[i]);
+
+    attr_index += (int) vendor_msg->p_vendor_data[7 + attr_index] + 8;
+  }
+}
```

Comment:
I suggest to remove the static buffer "meta_text_buffer". Instead, in the function "handle_rc_attributes_rsp", to "malloc" and "free" a 128Byte memory is enough, as the attribute text are transfered one by one for each attribute. Or we can use the p_vendor_data[] itself and thus the 128Byte malloc-ed buffer may not be necessary.
In this way we can save the static buffer, and the application layer is responsible to copy the data in the callback function, the example code should be changed accordingly(Implement a "deep copy" function used in bt_app_rc_ct_cb when invoke "bt_app_work_dispatch", and free the buffer after it is used.)